server/rating.py

```diff
@@ -10,36 +10,57 @@ class RatingType(Enum):
     GLOBAL = "global"
     LADDER_1V1 = "ladder_1v1"
 
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return self.value == other
+        else:
+            return super().__eq__(other)
 
-class RatingTypeMap(MutableMapping):
-    def __init__(self, default):
-        MutableMapping.__init__(self)
-        self._back = {}
+    def __hash__(self):
+        return hash(self.value)
 
-        for rtype in RatingType:
-            self[rtype] = default
 
-    def __getitem__(self, key: RatingType):
-        return self._back[key]
+K = Union[RatingType, str]
+V = TypeVar("V")
 
-    def __setitem__(self, key: RatingType, value):
-        self._back[key] = value
 
-    def __delitem__(self, key: RatingType):
-        del self._back[key]
+class RatingTypeMap(DefaultDict[K, V]):
+    """
+    A thin wrapper around `defaultdict` which stores RatingType keys as strings.
+    """
+    def __init__(self, default_factory, *args, **kwargs):
+        super().__init__(default_factory, *args, **kwargs)
 
-    def __iter__(self):
-        return iter(self._back)
+        # Initialize defaults for enumerated rating types
+        for rating in RatingType:
+            self[rating]
```

Comment:
nono! :-D

Reply:
don't you want to have `self[rating.value]` here?
```python
>>> from enum import Enum
>>> class RatingType(Enum):
...     GLOBAL = "global"
...     LADDER_1V1 = "ladder_1v1"
... 
>>> [key for key in RatingType]
[<RatingType.GLOBAL: 'global'>, <RatingType.LADDER_1V1: 'ladder_1v1'>]
>>> [key.value for key in RatingType]
['global', 'ladder_1v1']

```

Reply:
It can be either. That's the point of `RatingTypeMap`. Notice that `RatingType.LADDER_1V1 == "ladder_1v1"`.

Reply:
```python
>>> from enum import Enum
>>> class RatingType(Enum):
...     GLOBAL = "global"
...     LADDER_1V1 = "ladder_1v1"
... 
>>> x = {"ladder_1v1": "OK"}
>>> rt = RatingType.LADDER_1V1
>>> x[rt]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: <RatingType.LADDER_1V1: 'ladder_1v1'>
>>> x[rt.value]
'OK'

```

it's not the same.

Reply:
You did not copy the definition of `RatingType` correctly

Reply:
But I think I’m going to get rid of the enum anyways and just have class constants that are strings. Then it’s simpler